opencog/atomspace/Foreach.h

```diff
@@ -18,16 +18,16 @@ namespace opencog
  */
 
 
-//! Invoke the callback on each atom in the outgoing set of handle h.
+//! Invoke the callback on each atom in the outgoing set of handle h
+//! till one of them return true, and returns true as a
+//! result. Otherwise the callback is called on all outgoings and
+//! false is returned.
 /// XXX TODO this should be moved to class Link
```

Comment:
FYI, I've never been entirely thrilled with these foreach-callback tools, so if you ever glimpse a better way, and have the time to fiddle with it, it could be replaced...